client/client_with_secret.go

```diff
@@ -69,6 +70,12 @@ func (cws *WithSecret) TokenSource() oauth2.TokenSource {
 	return cws.credentials.TokenSource(context.Background())
 }
 
+// UMATokenSource returns a token source that returns a new token each time the Token()
```

Comment:
I think this comment is too verbose, I don't think we should explain in this comment what `UMATokenSource` does.

Reply:
Had a similar approach for the other use-cases as well, but I'm happy to simplify a bit.